x/cdp/keeper/cdp.go

```diff
@@ -187,7 +190,10 @@ func (k Keeper) GetCDP(ctx sdk.Context, collateralDenom string, cdpID uint64) (t
 // SetCDP sets a cdp in the store
 func (k Keeper) SetCDP(ctx sdk.Context, cdp types.CDP) {
 	store := prefix.NewStore(ctx.KVStore(k.key), types.CdpKeyPrefix)
-	db, _ := k.GetDenomPrefix(ctx, cdp.Collateral.Denom)
+	db, found := k.GetDenomPrefix(ctx, cdp.Collateral.Denom)
+	if !found {
+		panic(fmt.Sprintf("invalid collateral denom %s", cdp.Collateral.Denom))
+	}
```

Comment:
Consider using `ErrInvalidCollateral` instead of panic. If you want to avoid returning an error then put functionality inside `if found { ... }` statement.